core/src/test/scala/sttp/client4/testing/streaming/StreamingTest.scala

```diff
@@ -189,6 +191,17 @@ abstract class StreamingTest[F[_], S]
     val numChunks = 100
     val url = uri"https://httpbin.org/stream/$numChunks"
 
+    implicit val monadError: MonadError[Future] = new FutureMonad
+
+    def retryImmediatelyOnError[A](action: => Future[A], retriesLeft: Int): Future[A] =
+      action.handleError { case error =>
```

Comment:
**Nitpick**: do you need case here?

Reply:
Yes, at first I wrote without case, but needed to add it for compatibility with Scala.js https://github.com/softwaremill/sttp/pull/1874/commits/57724383e90475e41bc4748cfc7c0997efb9e67b